src/gui/DatabaseWidget.cpp

```diff
@@ -755,7 +755,7 @@ void DatabaseWidget::entryActivationSignalReceived(Entry* entry, EntryModel::Mod
 void DatabaseWidget::switchToEntryEdit()
 {
     Entry* entry = m_entryView->currentEntry();
-    Q_ASSERT(entry);
+    
```

Comment:
Actually, thinking more about it, switchToEntryEdit() already IS the intermediate member which catches invalid entries, because then it delegates further to switchToEntryEdit(Entry*, bool). Having an assertion there makes sense (and actually, there is one), but having an assertion already in switchToEntryEdit() is a mistake.